### PR TITLE
Conflict resolution / issues

### PR DESCRIPTION
**Pull request by tiltec**
Related to https://github.com/yunity/karrot-frontend/issues/853
https://community.foodsaving.world/t/conflict-resolution-with-possibility-to-remove-user-from-group/201

Open tasks:
- [ ] refine emails and possibly add more
  - [x] add emails to affected user
- [x] add custom title for push messages (to tell them apart from private messages)
- [x] test how it works from the perspective of the removed user
  - emails
  - notifications
  - can they view the case chat?
  - can they write in the chat after they have been removed?
- [x] tests: post message in case conversation
- [x] maybe only allow creating conflict resolution cases if there are more then x active editors
- [x] maybe don't add new editors to existing cases, might prevent some manipulation (needs concept of case membership)
- [x] throttle case creation to sane limit
- [x] increase test coverage

### CODE REVIEW

**Comment by tiltec** [comment]
The main response from the `case` API currently looks like this:

```
{'affected_user': 174,
 'created_at': '2019-01-10T13:56:24.556407Z',
 'created_by': 173,
 'group': 36,
 'id': 35,
 'is_decided': False,
 'topic': 'I complain about this user',
 'type': 'conflict_resolution',
 'votings': [{'accepted_option': 74,
              'expires_at': '2019-01-17T13:56:24.635931Z',
              'options': [{'affected_user': None,
                           'id': 73,
                           'mean_score': 1.0,
                           'message': None,
                           'type': 'further_discussion'},
                          {'affected_user': 174,
                           'id': 74,
                           'mean_score': 5.0,
                           'message': None,
                           'type': 'remove_user'}]}]}
```

@djahnie @taistadam

**Comment by tiltec** [comment]
updated case API response:

`GET /api/cases/:id/`
```py
{'affected_user': 4,
 'created_at': '2019-01-14T14:46:12.663092Z',
 'created_by': 3,
 'group': 2,
 'id': 2,
 'is_decided': False,
 'topic': 'I complain about this user',
 'type': 'conflict_resolution',
 'votings': [{'accepted_option': 7,
              'expires_at': '2019-01-21T14:46:12.756294Z',
              'id': 3,
              'options': [{'affected_user': None,
                           'id': 7,
                           'message': None,
                           'sum_score': 1.0,
                           'type': 'further_discussion',
                           'your_score': 1},
                          {'affected_user': None,
                           'id': 8,
                           'message': None,
                           'sum_score': 1.0,
                           'type': 'no_change',
                           'your_score': 1},
                          {'affected_user': 4,
                           'id': 9,
                           'message': None,
                           'sum_score': 1.0,
                           'type': 'remove_user',
                           'your_score': 1}],
              'participant_count': 1},
             {'accepted_option': None,
              'expires_at': '2019-01-28T15:46:12.906408Z',
              'id': 4,
              'options': [{'affected_user': None,
                           'id': 10,
                           'message': None,
                           'sum_score': None,
                           'type': 'further_discussion',
                           'your_score': None},
                          {'affected_user': None,
                           'id': 11,
                           'message': None,
                           'sum_score': None,
                           'type': 'no_change',
                           'your_score': None},
                          {'affected_user': 4,
                           'id': 12,
                           'message': None,
                           'sum_score': None,
                           'type': 'remove_user',
                           'your_score': None}],
              'participant_count': 0}]}
```

File: foodsaving/cases/templates/conflict_resolution_case_continued.mjml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        <mj-title>{% trans affected_user_name=affected_user.display_name|safe %}Continue the discussion about the conflict with {{ affected_user_name }}{% endtrans %}
+        </mj-title>
+    </mj-head>
+    <mj-body>
+        <mj-include path="../../base/templates/_header.mjml"/>
+        <mj-section background-color="#ffffff" padding="40px 0 40px 0">
+            <mj-column>
+                <mj-text>
+                    <h1>{% trans affected_user_name=affected_user.display_name|safe %}Continue the discussion about the conflict with {{ affected_user_name }}{% endtrans %}
+                    <p>{% trans %}The most acceptable outcome of the last voting round is to continue discussion and start another round of voting.{% endtrans %}</p>
```
Comment (djahnie) [inline foodsaving/cases/templates/conflict_resolution_case_continued.mjml]: 'The most acceptable outcome' sounds weird to me. I'd replace the phrase with 'The result of the last voting round is to continue discussion and start another round of voting.'

File: foodsaving/cases/templates/conflict_resolution_case_continued.mjml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                    <p>{% trans expiry_date=expires_at|date %}This voting round ends at {{ expiry_date }}.{% endtrans %}</p>
```
Comment (djahnie) [inline foodsaving/cases/templates/conflict_resolution_case_continued.mjml]: 'The new voting round' makes it clearer that it's about the one that just started.

File: foodsaving/cases/templates/conflict_resolution_case_continued.mjml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<mjml>
+    <mj-head>
+        <mj-include path="../../base/templates/_base-head.mjml"/>
+        <mj-title>{% trans affected_user_name=affected_user.display_name|safe %}Continue the discussion about the conflict with {{ affected_user_name }}{% endtrans %}
```
Comment (djahnie) [inline foodsaving/cases/templates/conflict_resolution_case_continued.mjml]: Why not 'Keep discussing'? It's shorter than 'Continue the discussion' and the same string as the one we use in the frontend.

File: foodsaving/cases/templates/conflict_resolution_case_continued.subject.jinja2 [hunk 4]
```diff
@@ -0,0 +1 @@
+[{{ site_name }}] 🔥 {% trans affected_user_name=affected_user.display_name|safe %}Continue the discussion about the conflict with {{ affected_user_name }}{% endtrans %}
```
Comment (djahnie) [inline foodsaving/cases/templates/conflict_resolution_case_continued.subject.jinja2]: 'Keep discussing'

File: foodsaving/cases/templates/new_conflict_resolution_case.mjml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<mjml>
+    <mj-head>
+        <mj-include path="../../base/templates/_base-head.mjml"/>
+        <mj-title>{% trans creator_name=created_by.display_name, affected_user_name=affected_user.display_name %}{{ creator_name }} opened a discussion about a conflict with {{ affected_user_name }}{% endtrans %}</mj-title>
```
Comment (djahnie) [inline foodsaving/cases/templates/new_conflict_resolution_case.mjml]: Hmm... I don't love it. But I also don't know if any of these are better:
- 'X started a conflict resolution process about Y'
- 'A conflict resolution process was started about Y by X'

File: foodsaving/cases/templates/new_conflict_resolution_case.mjml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    </mj-head>
+    <mj-body>
+        <mj-include path="../../base/templates/_header.mjml"/>
+        <mj-section background-color="#ffffff" padding="40px 0 40px 0">
+            <mj-column>
+                <mj-text>
+                    <h1>{% trans creator_name=created_by.display_name, affected_user_name=affected_user.display_name %}{{ creator_name }} opened a discussion about a conflict with {{ affected_user_name }}{% endtrans %}</h1>
```
Comment (djahnie) [inline foodsaving/cases/templates/new_conflict_resolution_case.mjml]: Same as above.

File: foodsaving/cases/templates/new_conflict_resolution_case.subject.jinja2 [hunk 7]
```diff
@@ -0,0 +1 @@
+[{{ site_name }}] 💣 {% trans creator_name=created_by.display_name|safe, affected_user_name=affected_user.display_name|safe %}{{ creator_name }} opened a discussion about a conflict with {{ affected_user_name }}{% endtrans %}
```
Comment (djahnie) [inline foodsaving/cases/templates/new_conflict_resolution_case.subject.jinja2]: Same as above.

File: foodsaving/conversations/api.py [hunk 8]
```diff
@@ -147,7 +149,13 @@ def list(self, request, *args, **kwargs):
             filter(id__in=[c.target_id for c in application_conversations]). \
             select_related('user')
 
-        # Applicant does not have access to group member profiles, so we sideload reduced user profiles
+        issues_ct = ContentType.objects.get_for_model(Issue)
+        issue_conversations = [item for item in conversations if item.target_type == issues_ct]
+        issues = Issue.objects. \
+            filter(id__in=[c.target_id for c in issue_conversations]). \
+            prefetch_stuff(user=request.user)
```
Comment (nicksellen) [inline foodsaving/conversations/api.py]: `_stuff`? ;) `_related`?

File: foodsaving/issues/__init__.py [hunk 9]
```diff
@@ -0,0 +1,8 @@
+from django.apps import AppConfig
+
+
+class CasesConfig(AppConfig):
```
Comment (nicksellen) [inline foodsaving/issues/__init__.py]: Cases!

**Pull request merged**